scalyr_agent/configuration.py

```diff
@@ -3603,6 +3603,19 @@ def __verify_workers_entry_and_set_defaults(self, worker_entry, entry_index=None
                 worker_entry, "api_key", self.api_key, description % entry_index
             )
         else:
+
+            # validate the worker id. It should consist only from this set of characters.
+            allowed_worker_id_characters_pattern = "a-zA-Z0-9_"
+            if re.search(
+                r"[^{0}]+".format(allowed_worker_id_characters_pattern), worker_id
```

Comment:
LGTM.

Eventually we could add this as a native "feature" for any string option, but this works for now.